libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1050,8 +1072,26 @@ void CompilerUtils::convertType(
 	}
 	case Type::Category::ArraySlice:
 	{
-		solAssert(_targetType.category() == Type::Category::Array, "");
 		auto& typeOnStack = dynamic_cast<ArraySliceType const&>(_typeOnStack);
+		if (_targetType.category() == Type::Category::FixedBytes)
+		{
+			solAssert(
+				typeOnStack.arrayType().isByteArray(),
+				"Array types other than bytes not convertible to bytesNN."
+			);
+			m_context << Instruction::SWAP1;
+			m_context.callYulFunction(
+				m_context.utilFunctions().bytesToBytesNNConversionFunction(
+					typeOnStack.arrayType(),
+					dynamic_cast<FixedBytesType const&>(_targetType)
+				),
+				(typeOnStack.dataStoredIn(DataLocation::CallData) && typeOnStack.isDynamicallySized()) ? 2 : 1,
```

Comment:
Can you compute that above the function call and then assert that it is equal to the stack size of `typeOnStack`?

Reply:
I added more assertions and made it much more explicit, I think it should be fine now